2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +20,14 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+  let shoppingList = {};
+  shoppingList.name = recipe.name;
+  shoppingList.items = [];
+  recipe.ingredients.map(item => {
+    if (!pantry.fridgeContents.includes(item) && !pantry.cupboardContents.includes(item))
+      shoppingList.items.push(item);
```

Comment:
great job, you can use concat function from Javascript to combine both arrays and have one array and check the items in that array ==> `let allIHave = pantry.fridgeContents.concat(cupBoardContents)`